src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -203,15 +208,25 @@ private async Task UpdateJobRecordAsync(CancellationToken cancellationToken)
                 SearchResult searchResult = null;
 
                 // Search and process the results.
-                using (IScoped<ISearchService> searchService = _searchServiceFactory())
+                switch (_exportJobRecord.ExportType)
                 {
-                    searchResult = await searchService.Value.SearchAsync(
-                        resourceType: null,
-                        queryParametersList,
-                        cancellationToken);
+                    case ExportJobType.All:
+                    case ExportJobType.Patient:
+                        using (IScoped<ISearchService> searchService = _searchServiceFactory())
+                        {
+                            searchResult = await searchService.Value.SearchAsync(
+                                resourceType: null,
+                                queryParametersList,
+                                cancellationToken);
+                        }
+
+                        break;
+                    case ExportJobType.Group:
+                        searchResult = await GetGroupPatients(_exportJobRecord.GroupId, queryParametersList, cancellationToken);
+                        break;
                 }
 
-                if (_exportJobRecord.ExportType == ExportJobType.Patient)
+                if (_exportJobRecord.ExportType != ExportJobType.All)
```

Comment:
nit: Might be better to check specifically for `Patient || Group`. In the future if we add or change an `ExportJobType` type and miss updating this condition, it might lead to unexpected behavior.